covid_microservice/api/router.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Desc: Module that handles the main API functionality.
+"""
+
+
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+from hexkit.protocols.dao import ResourceNotFoundError
+from hexkit.providers.mongodb.provider import MongoDbDaoNaturalId
+
+from ..core.models import NewSampleResponse, NewSampleSubmission, PcrTest, UpdatePcrTest
+from ..core.utils import make_access_token, make_sample_id
+from .deps import get_mongodb_pcrtest_dao
+
+MSG_NOT_FOUND = "Specified resource was not found."
+
+# This APIRouter instance will be referenced/included by 'app' in main.py
+sample_router = APIRouter()
+
+
+# GET /sample
+@sample_router.get(
+    "/sample/{access_token}",
```

Comment:
The access token should not be used as sample ID but as a bearer token in the header.

Reply:
Rest resources should be in plural.